src/data/mutable_data.rs

```diff
@@ -308,3 +318,251 @@ impl Debug for MutableData {
                self.owners)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    extern crate rand;
+
+    use rust_sodium::crypto::sign;
+    use std::collections::{BTreeMap, BTreeSet};
+    use std::iter;
+    use error::RoutingError;
+    use super::*;
+
+    macro_rules! assert_err {
+        ($left: expr, $err: path) => {{
+            assert!(if let Err($err) = $left { true } else { false });
```

Comment:
This will not give much idea on failure - it will only say something asserted to false. If you want to write it that way ^^^ it's better to have something like:
```
match $left {
   Err($err) => (),
   x => panic!("Expected: {:?}  ### Found: {:?}", $err, x),
}
```